SimplenoteUITests/Settings.swift

```diff
@@ -0,0 +1,48 @@
+import XCTest
+
+let screenName = "Settings"
+
+enum SettingsScreen: String {
+
+    case doneButton = "Done"
+    case condensedModeSwitch = "Condensed Note List"
+
+    var element: XCUIElement {
+        switch self {
+        case .doneButton:
+            return app.navigationBars[screenName].buttons[self.rawValue]
+        case .condensedModeSwitch:
+            return app.tables.cells.containing(.staticText, identifier: self.rawValue).firstMatch
+        }
+    }
+}
+
+class Settings {
+
+    static func condensedModeEnable() {
+        let condensedModeSwitch = SettingsScreen.condensedModeSwitch.element
+        let switchValue = condensedModeSwitch.value as? String
+
+        if switchValue == "0" {
+            condensedModeSwitch.tap()
+        }
+    }
+
+    static func condensedModeDisable() {
+        let condensedModeSwitch = SettingsScreen.condensedModeSwitch.element
+        let switchValue = condensedModeSwitch.value as? String
+
+        if switchValue == "1" {
+            condensedModeSwitch.tap()
+        }
+    }
```

Comment:
Nitpick. You could DRY this code with something like

```swift
static func condensedModeDisable() {
  switchCondensedModeIfMatching(value: "1")
}

static private func switchCondensedModeIfMatching(value: String) {
    let condensedModeSwitch = SettingsScreen.condensedModeSwitch.element
    let switchValue = condensedModeSwitch.value as? String

    guard switchValue == value else { return }

    condensedModeSwitch.tap()
}
```

In the context of a screen object like this, having as much code that introspects the UI DRY is valuable because when know UI changes often and centralization lowers the cost of updating our tests.

Reply:
Oh I feel bad for not doing this 🤦. Finding more opportunities for code reuse was my favourite, whenever I was reviewing the automated tests before at my previous place. And I fell into this trap myself. Thank you!

Reply:
I've moved the code to another more generic `toggleSwitchIfNeeded(_ switchElement: XCUIElement, _ value: String)` which I placed in `Generic.swift` for now, because it'll be useful for the other switched too. So now it looks like this:

```
    static func condensedModeEnable() {
        switchCondensedModeIfNeeded(value: "1")
    }

    static func condensedModeDisable() {
        switchCondensedModeIfNeeded(value: "0")
    }

    static func switchCondensedModeIfNeeded(value: String) {
        toggleSwitchIfNeeded(SettingsScreen.condensedModeSwitch.element, value)
    }

...

func toggleSwitchIfNeeded(_ switchElement: XCUIElement, _ value: String) {
    let switchValue = switchElement.value as? String

    guard switchValue != value else { return }

    switchElement.tap()
}
```

Reply:
Nice!